crates/matrix-sdk-common/src/deserialized_responses.rs

```diff
@@ -181,14 +195,40 @@ pub enum DeviceLinkProblem {
 pub enum ShieldState {
     /// A red shield with a tooltip containing the associated message should be
     /// presented.
-    Red { message: &'static str },
+    Red {
+        /// A machine-readable representation.
+        code: ShieldStateCode,
+        /// A human readable description.
+        message: &'static str,
+    },
     /// A grey shield with a tooltip containing the associated message should be
     /// presented.
-    Grey { message: &'static str },
+    Grey {
+        /// A machine-readable representation.
+        code: ShieldStateCode,
+        /// A human readable description.
+        message: &'static str,
+    },
     /// No shield should be presented.
     None,
 }
 
+/// A machine-readable representation of the authenticity for a `ShieldState`.
+#[derive(Clone, Debug, Deserialize, Serialize, Eq, PartialEq)]
+#[cfg_attr(feature = "uniffi", derive(uniffi::Enum))]
+pub enum ShieldStateCode {
```

Comment:
If some client app persist that and then later we add a new code, how will this behave? When deserializing an old version of this enum?

Reply:
I'm no Rust expert but the below seems to indicate that it will be fine if serialised to JSON:

<details>
<summary>Example written with serde</summary>
```
use serde::{Serialize, Deserialize};
use serde_json;

#[derive(Serialize, Deserialize, Debug)]
enum Color {
    Red,
}


#[derive(Serialize, Deserialize, Debug)]
enum ColorPop {
    Red,
    Green,
}


fn main() {
    let color = Color::Red;
    let json_string = serde_json::to_string(&color).unwrap();
    println!("Serialized color: {}", json_string); // Serialized color: "Red"
    
    let color: ColorPop = serde_json::from_str(&json_string).unwrap();
    println!("Deserialized color: {:?}", color); // Deserialized color: Red
}
```
</details>

How come the ShieldState is serialisable in the first place? Shouldn't the app always ask the SDK for the state to be up to date?

Reply:
It might be for the WASM use-case where we sometimes, instead of adding a new type in the bindings which the SDK type gets converted into, we return a JSON object. This is using [serde-wasm-bindgen](https://docs.rs/serde-wasm-bindgen/latest/serde_wasm_bindgen/).

If that's not the case, and nobody needs the implementations, perhaps we should just remove the `Serialize` and `Deserialize` implementations since they might have been cargo culted.

Reply:
Got it, I didn't know that about WASM 👍